conformance/lang/expressions/transactional-expr/transactional_expr.balt

```diff
@@ -0,0 +1,558 @@
+        io:println(transactional); // output true
+        check commit;
+        io:println(transactional); // output false
+    }
+}
+
+transactional function f1() {
+    io:println(transactional);
+}
+
+Test-Case: output
+Description: Test whether the transactional expression is being evaluated in transaction mode.
+Labels: transactional-expr, check, optional-type, error, transaction-stmt, commit-action, function-call-expr,
+        transactional-function
+
+function init() returns error? {
+    transactional function () trxFunc = transactional function() {
+        io:println(transactional);
+    };
+
+    transaction {
+        trxFunc(); // output true
+        io:println(transactional); // output true
+        check commit;
+        io:println(transactional); // output false
+    }
+}
+
+Test-Case: output
+Description: Test whether the transactional expression is being evaluated in transaction mode.
+Labels: transactional-expr, check, optional-type, error, transaction-stmt, commit-action, transaction:onCommit,
+        isolated-function, var, explicit-anonymous-function-expr, transaction:Info
+
+function init() returns error? {
+    var onCommitFunc = isolated function(transaction:Info? info) {
+        io:println(transactional);
+    };
+
+    transaction {
+        transaction:onCommit(onCommitFunc); // output true
+        io:println(transactional); // output true
+        check commit;
+        io:println(transactional); // output false
+    }
+}
+
+Test-Case: output
+Description: Test whether the transactional expression is being evaluated in transaction mode.
+Labels: transactional-expr, check, optional-type, error, int, transaction-stmt, rollback-stmt, commit-action,
+        transaction:onRollback, var, explicit-anonymous-function-expr, isolated-function, transaction:Info, is-expr,
+        if-else-stmt
+
+function init() returns error? {
+    var onRollbackFunc = isolated function(transaction:Info? info, error? cause, boolean willTry) {
+        io:println(transactional);
+    };
+
+    transaction {
+        int? a = 10;
+
+        if a is int {
+            io:println(transactional); // output true
+            transaction:onRollback(onRollbackFunc); // output true
+            io:println(transactional); // output true
+            rollback;
+            io:println(transactional); // output false
+        } else {
+            check commit;
+        }
+
+        io:println(transactional); // output false
+    }
+}
+
+Test-Case: output
+Description: Test whether the transactional expression is being evaluated in transaction mode.
+Labels: transactional-expr, check, optional-type, error, int, transaction-stmt, commit-action, transactional-worker
+
+function init() returns error? {
+    transaction {
+        f1();
+        check commit;
+        io:println(transactional); // output false
+    }
+}
+
+transactional function f1() {
+    transactional worker A {
+        io:println(transactional); // output true
+    }
+}
+
```

Comment:
Shall we add a test with non-transactional func invocation and checking transaction in the invoked function?